staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -599,6 +625,47 @@ func (s preparedGenericAPIServer) Run(stopCh <-chan struct{}) error {
 		s.NonLongRunningRequestWaitGroup.Wait()
 	}()
 
+	// wait for all in-flight watches to finish
+	activeWatchesDrainedCh := make(chan struct{})
+	go func() {
+		defer close(activeWatchesDrainedCh)
+
+		<-notAcceptingNewRequestCh.Signaled()
+		if s.ShutdownWatchTerminationGracePeriod == time.Duration(0) {
+			klog.V(1).InfoS("[graceful-termination] not going to wait for active watch request(s) to drain")
+			return
+		}
+
+		// Wait for all active watches to finish
+		grace := s.ShutdownWatchTerminationGracePeriod
+		activeBefore, activeAfter, err := s.WatchRequestWaitGroup.Wait(func(count int) (utilwaitgroup.RateLimiter, context.Context, context.CancelFunc) {
+			// TODO: this is back of the envelope for now
+			seconds := int(grace.Seconds())
+			if seconds <= 0 {
+				seconds = 2
+			}
+			qps := count / seconds
+			if qps < 200 {
+				qps = 200
+			}
```

Comment:
Why do we need to translate to seconds? I would say that any non-negative value is actually fine.

So I would just do:
```
qps := float64(count) / grace.Seconds()
if qps < 200 {
  qps = 200
}
```

Reply:
yes, changed it.